syntax = "proto3";

package gis.protobuf;

option java_package = "com.github.tomi77.gis";
option java_multiple_files = true;

// type: Point
// coordinates: [x, y]
message Point2D {
  float x = 1;
  float y = 2;
}

// type: Point
// coordinates: [x, y, z]
message Point3D {
  float x = 1;
  float y = 2;
  float z = 3;
}

// type: MultiPoint
// coordinates: [[x, y], [x, y]]
message MultiPoint2D {
    repeated Point2D point = 1;
}

// type: MultiPoint
// coordinates: [[x, y, z], [x, y, z]]
message MultiPoint3D {
    repeated Point3D point = 1;
}
